### PR TITLE
Fork_config loses previous_ label

### LINKED ISSUE #15070: The output of state export is not consistent in "previous_state_hash" and "previous_height"
The generated fork state contains for the following field:
```
    "fork": {
      "previous_state_hash": "3NLiejEBoNuXjgC59Jt8PUvvqNbPxgNFZBU4a9ZgrpNzzdXPYPbb",
      "previous_length": 368,
      "previous_global_slot": 2326
    }
```
The length and global_slot of "previous_state_hash" is different from what we have here.
It turns out that the "previous_length" and "previous_global_slot" corresponds to next block's length and global_slot.

### PR DESCRIPTION
**Pull request by emberian**
Closes #15070

Berkeley: https://github.com/MinaProtocol/mina/pull/15079

(the berkeley version is compatible with develop, unlike this one)

### CODE REVIEW

**Comment by emberian** [comment]
!ci-build-me

File: src/config/fork.mlh [hunk 1]
```diff
@@ -1,3 +1,3 @@
-[%%undef fork_previous_length]
+[%%undef fork_blockchain_length]
 [%%undef fork_previous_state_hash]
```
Comment (emberian) [inline src/config/fork.mlh:2]: missed a line...

**Comment by emberian** [comment]
!ci-build-me

**Review by ghost-not-in-the-shell — APPROVED** [review]
The change to runtime_config looks good. It's using the current state_hash instead of previous_state_hash. This should fix the bug we saw in the hf testnet.

**Comment by emberian** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**